packages/browser/src/lib/create-deferred.ts

```diff
@@ -4,13 +4,18 @@
 export const createDeferred = <T>() => {
   let resolve!: (value: T | PromiseLike<T>) => void
   let reject!: (reason: any) => void
+  let settled = false
   const promise = new Promise<T>((_resolve, _reject) => {
     resolve = _resolve
     reject = _reject
   })
+
+  promise.finally(() => (settled = true))
+
   return {
     resolve,
     reject,
     promise,
+    settled,
```

Comment:
This doesn't look like it actually works as expected, as `settled` is a value and not a function that is closing over the value. But rather than fixing it, maybe we can do without adding this to the API (per @oscb comments below?)

Reply:
very very good catch! although this will be removed since @oscb suggestion voids the purpose anyway